File: SoftUni Lessons/Python Development/Python Basics April 2019/Past Exams and Problems/9 - 20 and 21 April 2019/04. Easter Shop.py
owned_eggs = int(input())
sold_eggs = owned_eggs

total_sold_eggs = 0
command = None
count = 0

while command != "Close" or not(sold_eggs < 0):
    command = input()
    if command == "Close":
        break
    count = int(input())
    if command == "Fill":
        sold_eggs += count
    if command == "Buy":
        sold_eggs -= count
        total_sold_eggs += count
    if sold_eggs < 0:
        break

if command == "Close":
    print("Store is closed!")
    print(f"{total_sold_eggs} eggs sold.")

if sold_eggs < 0:
    print("Not enough eggs in store!")
    print(f"You can buy only {sold_eggs + count}.")

File: SoftUni Lessons/JavaScript Development/JavaScript Basics March 2020/21. МИНАЛИ ИЗПИТИ/01. Online Exam - 20 and 21 April 2019/04. Easter Shop.js
function easterShop(input) {
    // Mask
    let eggs = Number(input.shift());
    let sold = 0;

    while ("I say so!") {
        let buyFillClose = input.shift();
        
        if (buyFillClose == 'Close') {
            console.log(`Store is closed!`);
            console.log(`${sold} eggs sold.`)
            break;
        } else if (buyFillClose == "Fill") {
            eggs += Number(input.shift());
        } else if (buyFillClose == 'Buy') {
            let count = Number(input.shift());
            if (eggs - count < 0) {
                console.log(`Not enough eggs in store!`);
                console.log(`You can buy only ${eggs}.`);
                break;
            } else {
                eggs -= count;
                sold += count;
            }
        }
    }
}

easterShop([13, 'Buy', 8, 'Fill', 3, 'Buy', 10 ]);  // Should return:
// Not enough eggs in store!
// You can buy only 8.

easterShop([20, 'Fill', 30, 'Buy', 15, 'Buy', 20, 'Close']);  // Should return:
// Store is closed!
// 35 eggs sold.
